File: python_interface/tests/linexpr0_test.py
from elina_auxiliary_imports import *
from elina_linexpr0 import *
from tests.test_imports import *


def test_set_linexpr_scalar_int(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = 1

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    num = c_int(random.randint(0, 99))
    elina_linexpr0_set_cst_scalar_int(linexpr, num)
    for i in range(new_size.value):
        num = c_int(random.randint(0, 99))
        elina_linexpr0_set_coeff_scalar_int(linexpr, ElinaDim(i), num)

    print_c('set scalar int size : {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_set_linexpr_scalar_frac(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = c_size_t(1)

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    p = c_long(random.randint(0, 999))
    q = c_ulong(random.randint(1, 20))
    elina_linexpr0_set_cst_scalar_frac(linexpr, p, q)

    for i in range(new_size.value):
        p = c_long(random.randint(0, 999))
        q = c_ulong(random.randint(1, 20))
        elina_linexpr0_set_coeff_scalar_frac(linexpr, ElinaDim(i), p, q)

    print_c('set scalar frac size: {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_set_linexpr_scalar_double(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = 1

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    d = c_double(random.uniform(-1, 1))
    elina_linexpr0_set_cst_scalar_double(linexpr, d)
    for i in range(new_size.value):
        d = c_double(random.uniform(-1, 1))
        elina_linexpr0_set_coeff_scalar_double(linexpr, ElinaDim(i), d)

    print_c('set scalar double size : {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_set_linexpr_interval_int(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = 1

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    inf = c_int(random.randint(0, 99))
    sup = c_int(random.randint(inf.value, 99 + inf.value))
    elina_linexpr0_set_cst_interval_int(linexpr, inf, sup)
    for i in range(new_size.value):
        inf = c_int(random.randint(0, 99))
        sup = c_int(random.randint(inf.value, 99 + inf.value))
        elina_linexpr0_set_coeff_interval_int(linexpr, ElinaDim(i), inf, sup)

    print_c('set interval int size : {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_set_linexpr_interval_frac(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = 1

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    numinf = c_long(random.randint(0, 99))
    deninf = c_ulong(random.randint(1, 20))
    numsup = c_long(numinf.value + random.randint(0, 99))
    densup = deninf
    elina_linexpr0_set_cst_interval_frac(linexpr, numinf, deninf, numsup, densup)
    for i in range(new_size.value):
        numinf = c_long(random.randint(0, 99))
        deninf = c_ulong(random.randint(1, 20))
        numsup = c_long(numinf.value + random.randint(0, 99))
        densup = deninf
        elina_linexpr0_set_coeff_interval_frac(linexpr, ElinaDim(i), numinf, deninf, numsup, densup)

    print_c('set interval frac size : {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_set_linexpr_interval_double(linexpr):

    size = elina_linexpr0_size(linexpr)
    new_size = size // 2

    if not new_size:
        new_size = 1

    new_size = c_size_t(new_size)
    elina_linexpr0_realloc(linexpr, new_size)

    inf = c_double(random.uniform(-1, 1))
    sup = c_double(inf.value + random.uniform(-1, 1))
    elina_linexpr0_set_cst_interval_double(linexpr, inf, sup)
    for i in range(new_size.value):
        inf = c_double(random.uniform(-1, 1))
        sup = c_double(inf.value + random.uniform(-1, 1))
        elina_linexpr0_set_coeff_interval_double(linexpr, ElinaDim(i), inf, sup)

    print_c('set interval double size : {} linexpr: '.format(new_size))
    elina_linexpr0_print(linexpr, None)
    print_c(' is linear: {} is quasilinear: {}\n'.format(elina_linexpr0_is_linear(linexpr),
                                                         elina_linexpr0_is_quasilinear(linexpr)))
    intdim = c_size_t(random.randint(0, new_size.value-1))
    realdim = c_size_t(random.randint(0, new_size.value-1))
    print_c('dim: {} is integer: {} dim: {} is real: {}\n'.format(intdim, elina_linexpr0_is_integer(linexpr, intdim),
                                                                  realdim, elina_linexpr0_is_real(linexpr, realdim)))


def test_linexpr_compare(linexpr1, linexpr2, linexpr3, linexpr4):

    print_c('linexpr1: ')
    elina_linexpr0_print(linexpr1, None)

    print_c('\nlinexpr2: ')
    elina_linexpr0_print(linexpr2, None)

    print_c('\nlinexpr3: ')
    elina_linexpr0_print(linexpr3, None)

    print_c('\nlinexpr4: ')
    elina_linexpr0_print(linexpr4, None)

    print_c('\nlinexpr1 <= linexpr2: {} linexpr2 <= linexpr1: {}'.format(elina_linexpr0_compare(linexpr1, linexpr2),
                                                                         elina_linexpr0_compare(linexpr2, linexpr1)))

    print_c('\nlinexpr1 <= linexpr3: {} linexpr3 <= linexpr1: {}'.format(elina_linexpr0_compare(linexpr1, linexpr3),
                                                                         elina_linexpr0_compare(linexpr3, linexpr1)))

    print_c('\nlinexpr1 <= linexpr4: {} linexpr4 <= linexpr1: {}'.format(elina_linexpr0_compare(linexpr1, linexpr4),
                                                                         elina_linexpr0_compare(linexpr4, linexpr1)))

    print_c('\nlinexpr2 <= linexpr3: {} linexpr3 <= linexpr2: {}'.format(elina_linexpr0_compare(linexpr2, linexpr3),
                                                                         elina_linexpr0_compare(linexpr3, linexpr2)))

    print_c('\nlinexpr2 <= linexpr4: {} linexpr4 <= linexpr2: {}'.format(elina_linexpr0_compare(linexpr2, linexpr4),
                                                                         elina_linexpr0_compare(linexpr4, linexpr2)))

    print_c('\nlinexpr3 <= linexpr4: {} linexpr4 <= linexpr3: {}\n'.format(elina_linexpr0_compare(linexpr3, linexpr4),
                                                                         elina_linexpr0_compare(linexpr4, linexpr3)))


def test_linexpr_equality(linexpr1, linexpr2, linexpr3, linexpr4):

    print_c('linexpr1: ')
    elina_linexpr0_print(linexpr1, None)

    print_c('\nlinexpr2: ')
    elina_linexpr0_print(linexpr2, None)

    print_c('\nlinexpr3: ')
    elina_linexpr0_print(linexpr3, None)

    print_c('\nlinexpr4: ')
    elina_linexpr0_print(linexpr4, None)

    print_c('\nlinexpr1 == linexpr2: {}'.format(elina_linexpr0_equal(linexpr1, linexpr2)))
    print_c('\nlinexpr1 == linexpr3: {}'.format(elina_linexpr0_equal(linexpr1, linexpr3)))
    print_c('\nlinexpr1 == linexpr4: {}'.format(elina_linexpr0_equal(linexpr1, linexpr4)))
    print_c('\nlinexpr2 == linexpr3: {}'.format(elina_linexpr0_equal(linexpr2, linexpr3)))
    print_c('\nlinexpr2 == linexpr4: {}'.format(elina_linexpr0_equal(linexpr2, linexpr4)))
    print_c('\nlinexpr3 == linexpr4: {}\n'.format(elina_linexpr0_equal(linexpr3, linexpr4)))

size = c_size_t(random.randint(3, 22))
linexpr1 = elina_linexpr0_alloc(ElinaLinexprDiscr.ELINA_LINEXPR_SPARSE, size)
linexpr2 = elina_linexpr0_alloc(ElinaLinexprDiscr.ELINA_LINEXPR_DENSE, size)
linexpr3 = elina_linexpr0_alloc(ElinaLinexprDiscr.ELINA_LINEXPR_SPARSE, size)
linexpr4 = elina_linexpr0_alloc(ElinaLinexprDiscr.ELINA_LINEXPR_SPARSE, size)

test_set_linexpr_scalar_int(linexpr1)
test_set_linexpr_scalar_int(linexpr2)

test_set_linexpr_scalar_frac(linexpr1)
test_set_linexpr_scalar_frac(linexpr2)

test_set_linexpr_scalar_double(linexpr1)
test_set_linexpr_scalar_double(linexpr2)

test_set_linexpr_interval_int(linexpr3)
test_set_linexpr_interval_int(linexpr4)

test_set_linexpr_interval_frac(linexpr3)
test_set_linexpr_interval_frac(linexpr4)

test_set_linexpr_interval_double(linexpr3)
test_set_linexpr_interval_double(linexpr4)

test_linexpr_compare(linexpr1, linexpr2, linexpr3, linexpr4)
test_linexpr_equality(linexpr1, linexpr2, linexpr3, linexpr4)

File: python_interface/tests/linexpr0_test.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "elina_linexpr0.h"


void test_set_linexpr_scalar_int(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	int num = rand()%100;
	elina_linexpr0_set_cst_scalar_int(linexpr,num);
	for(i=0; i < new_size; i++){
		num = rand()%100;
		elina_linexpr0_set_coeff_scalar_int(linexpr,i,num);
	}
	printf("set scalar int size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%new_size;
	size_t realdim = rand()%new_size;
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}


void test_set_linexpr_scalar_frac(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	
	long int p = rand()%1000;
	unsigned long int q= rand()%20+1;
	elina_linexpr0_set_cst_scalar_frac(linexpr,p,q);
	for(i=0; i < new_size; i++){
		p = rand()%1000;
		q= rand()%20+1;
		elina_linexpr0_set_coeff_scalar_frac(linexpr,i,p,q);
	}
	printf("set scalar frac size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%(new_size+1);
	size_t realdim = rand()%(new_size+1);
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}


void test_set_linexpr_scalar_double(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	double d = (double)rand()/RAND_MAX*2.0-1.0;
	elina_linexpr0_set_cst_scalar_double(linexpr,d);
	for(i=0; i < new_size; i++){
		d = (double)rand()/RAND_MAX*2.0-1.0;
		elina_linexpr0_set_coeff_scalar_double(linexpr,i,d);
	}
	printf("set scalar double size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%new_size;
	size_t realdim = rand()%new_size;
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}



void test_set_linexpr_interval_int(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	int inf = rand()%100;
	int sup = inf + rand()%100;
	elina_linexpr0_set_cst_interval_int(linexpr,inf,sup);
	for(i=0; i < new_size; i++){
		inf = rand()%100;
		sup = inf + rand()%100;
		elina_linexpr0_set_coeff_interval_int(linexpr,i,inf,sup);
	}

	printf("set interval int size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%new_size;
	size_t realdim = rand()%new_size;
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}



void test_set_linexpr_interval_frac(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	int numinf = rand()%100;
	unsigned short int deninf = rand()%20+1;	

	int numsup = numinf + rand()%100;
	unsigned short int densup = deninf;
	elina_linexpr0_set_cst_interval_frac(linexpr,numinf,deninf,numsup,densup);
	for(i=0; i < new_size; i++){
		numinf = rand()%100;
		deninf = rand()%20+1;
		
		numsup = numinf + rand()%100;
		densup = deninf;

		elina_linexpr0_set_coeff_interval_frac(linexpr,i,numinf,deninf,numsup,densup);
	}

	printf("set interval frac size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%new_size;
	size_t realdim = rand()%new_size;
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}


void test_set_linexpr_interval_double(elina_linexpr0_t *linexpr){
	size_t size = elina_linexpr0_size(linexpr);
	size_t new_size = size/2;
	if(!new_size){
		new_size = 1;
	}
	elina_linexpr0_realloc(linexpr,new_size);
	size_t i;
	double inf = (double)rand()/RAND_MAX*2.0-1.0;
	double sup = inf + (double)rand()/RAND_MAX*2.0-1.0;
	elina_linexpr0_set_cst_interval_double(linexpr,inf,sup);
	for(i=0; i < new_size; i++){
		inf = (double)rand()/RAND_MAX*2.0-1.0;
		sup = inf + (double)rand()/RAND_MAX*2.0-1.0;
		elina_linexpr0_set_coeff_interval_double(linexpr,i,inf,sup);
	}

	printf("set interval double size : %ld linexpr: ",new_size);
	elina_linexpr0_print(linexpr,NULL);
	printf(" is linear: %d is quasilinear: %d\n", elina_linexpr0_is_linear(linexpr), elina_linexpr0_is_quasilinear(linexpr));
	size_t intdim = rand()%new_size;
	size_t realdim = rand()%new_size;
	printf("dim: %zu is integer: %d dim: %zu is real: %d\n",intdim,elina_linexpr0_is_integer(linexpr,intdim),realdim,elina_linexpr0_is_real(linexpr,realdim));
}




void test_linexpr_compare(elina_linexpr0_t *linexpr1, elina_linexpr0_t *linexpr2,
			  elina_linexpr0_t *linexpr3, elina_linexpr0_t *linexpr4){

	printf("linexpr1: ");
	elina_linexpr0_print(linexpr1,NULL);
	printf("\n");

	printf("linexpr2: ");
	elina_linexpr0_print(linexpr2,NULL);
	printf("\n");
	
	printf("linexpr3: ");
	elina_linexpr0_print(linexpr3,NULL);
	printf("\n");

	printf("linexpr4: ");
	elina_linexpr0_print(linexpr4,NULL);
	printf("\n");

	printf(" linexpr1<=linexpr2: %d linexpr2<=linexpr1: %d\n",elina_linexpr0_compare(linexpr1,linexpr2), elina_linexpr0_compare(linexpr2,linexpr1));
	printf(" linexpr1<=linexpr3: %d linexpr3<=linexpr1: %d\n",elina_linexpr0_compare(linexpr1,linexpr3), elina_linexpr0_compare(linexpr3,linexpr1));
	printf(" linexpr1<=linexpr4: %d linexpr4<=linexpr1: %d\n",elina_linexpr0_compare(linexpr1,linexpr4), elina_linexpr0_compare(linexpr4,linexpr1));


	printf(" linexpr2<=linexpr3: %d linexpr3<=linexpr2: %d\n",elina_linexpr0_compare(linexpr2,linexpr3), elina_linexpr0_compare(linexpr3,linexpr3));
	printf(" linexpr2<=linexpr4: %d linexpr4<=linexpr2: %d\n",elina_linexpr0_compare(linexpr2,linexpr4), elina_linexpr0_compare(linexpr4,linexpr2));
	
	printf(" linexpr3<=linexpr4: %d linexpr4<=linexpr3: %d\n",elina_linexpr0_compare(linexpr3,linexpr4), elina_linexpr0_compare(linexpr4,linexpr3));
	
}


void test_linexpr_equality(elina_linexpr0_t *linexpr1, elina_linexpr0_t *linexpr2,
			  elina_linexpr0_t *linexpr3, elina_linexpr0_t *linexpr4){

	printf("linexpr1: ");
	elina_linexpr0_print(linexpr1,NULL);
	printf("\n");

	printf("linexpr2: ");
	elina_linexpr0_print(linexpr2,NULL);
	printf("\n");
	
	printf("linexpr3: ");
	elina_linexpr0_print(linexpr3,NULL);
	printf("\n");

	printf("linexpr4: ");
	elina_linexpr0_print(linexpr4,NULL);
	printf("\n");

	printf(" linexpr1==linexpr2: %d\n",elina_linexpr0_equal(linexpr1,linexpr2));
	printf(" linexpr1==linexpr3: %d\n",elina_linexpr0_equal(linexpr1,linexpr3));
	printf(" linexpr1==linexpr4: %d\n",elina_linexpr0_equal(linexpr1,linexpr4));


	printf(" linexpr2==linexpr3: %d \n",elina_linexpr0_equal(linexpr2,linexpr3));
	printf(" linexpr2==linexpr4: %d \n",elina_linexpr0_equal(linexpr2,linexpr4));
	
	printf(" linexpr3==linexpr4: %d \n",elina_linexpr0_equal(linexpr3,linexpr4));
	
}



int main(){
	srand (time(NULL));
	size_t size = rand()%20+3;

	elina_linexpr0_t * linexpr1 = elina_linexpr0_alloc(ELINA_LINEXPR_SPARSE,size);
	elina_linexpr0_t * linexpr2 = elina_linexpr0_alloc(ELINA_LINEXPR_DENSE,size);
	elina_linexpr0_t * linexpr3 = elina_linexpr0_alloc(ELINA_LINEXPR_SPARSE,size);
	elina_linexpr0_t * linexpr4 = elina_linexpr0_alloc(ELINA_LINEXPR_SPARSE,size);

	// create sparse and dense linexpr of type scalar int
	test_set_linexpr_scalar_int(linexpr1);
	fflush(stdout);
	test_set_linexpr_scalar_int(linexpr2);
	fflush(stdout);
	
	//create sparse and dense linexpr of type scalar fraction of two ints
	test_set_linexpr_scalar_frac(linexpr1);
	fflush(stdout);
	test_set_linexpr_scalar_frac(linexpr2);
	fflush(stdout);
	
	//create sparse and dense linexpr of type scalar double
	test_set_linexpr_scalar_double(linexpr1);
	fflush(stdout);
	test_set_linexpr_scalar_double(linexpr2);
	fflush(stdout);

	// create sparse and dense linexpr of type interval int
	test_set_linexpr_interval_int(linexpr3);
	fflush(stdout);
	test_set_linexpr_interval_int(linexpr4);
	fflush(stdout);

	
	//create sparse and dense linexpr of type interval fraction of two ints
	test_set_linexpr_interval_frac(linexpr3);
	fflush(stdout);
	test_set_linexpr_interval_frac(linexpr4);
	fflush(stdout);
	
	//create sparse and dense linexpr of type interval double
	test_set_linexpr_interval_double(linexpr3);
	fflush(stdout);
	test_set_linexpr_interval_double(linexpr4);
	fflush(stdout);

	
	//test for comparison and equality 
	test_linexpr_compare(linexpr1, linexpr2, linexpr3, linexpr4);
	fflush(stdout);
	test_linexpr_equality(linexpr1, linexpr2, linexpr3, linexpr4);
	fflush(stdout);
	

	elina_linexpr0_free(linexpr1);
	elina_linexpr0_free(linexpr2);
	elina_linexpr0_free(linexpr3);
	elina_linexpr0_free(linexpr4);
}
